### ISSUE TITLE
Seal member if possible

### ISSUE THREAD

**Issue by giggio** [code-cracker/code-cracker#372]
This is just like #371, but applies to methods that are virtual and inherited from a base class.

````csharp
class Base
{
    public virtual void Foo() { }
}
class C : Base
{
    public override void Foo() { }
}
````

Becomes:

````csharp
class Base
{
    public virtual void Foo() { }
}
class C : Base
{
    public override sealed void Foo() { }
}
````

An analysis has to be made if the method can me made virtual. The analyzer has to check if there is derived class, and if in those classes if there is an override of the method.

Category: `Performance`
Diagnostic Id: `CC0094`
Severity: `Hidden`

Due to the problems on #293 this is the suggested strategy for this issue:
1. Offer the diagnostic on the class declaration only, not anywhere else. It is a simple marker/hidden diagnostic, no further analysis needed.
2. The analysis is completed on the code fix provider. It checks, asynchronously, if there are classes overriding that member. If not, the code fix is offer. Otherwise it is not.

It has to check for cancellation frequently, as this analysis made on the code fix provider will be costly.

**Comment by giggio** [comment]
There is a problem in this analyzer, as we need to check on compilation end only if there are derived classes. I am not sure the analyzer is worth it and we already have other issues blocked because of this.

**Comment by adraut** [comment]
This also on the list of not really worth it in my opinion.  Marking things sealed can make unit testing complicated because it makes all the easy ways to override behavior useless.

**Comment by giggio** [comment]
Maybe we could make it an info, or deliver it disabled. I see value in it. But the problem is the same one from #293.

**Comment by giggio** [comment]
I am closing this issue because of the problems stated on #293.

**Comment by giggio** [comment]
I think I might have found a way to fix this that overcomes the problems stated on #293. We change it to a hidden diagnostic, and only offer a fix if we can, the code fix provider verifies if it is possible. I will reopen it and update the description.

**Comment by giggio** [comment]
I will take this one, as a proof of concept. If it works, we can look into the others that had the same problem.

**Comment by giggio** [comment]
Closed on #523
